src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java

```diff
@@ -455,11 +455,14 @@ private static Connection getClosedConnection()
             public Object invoke(Object proxy, Method method, Object[] args) throws Throwable
             {
                final String methodName = method.getName();
-               if ("abort".equals(methodName)) { 
+               if ("isValid".equals(methodName)) {
+                  return Boolean.FALSE;
+               }
+               else if ("close".equals(methodName)) { 
```

Comment:
Neither ``isClosed()`` nor ``isValid()`` implies an alteration of state, they are read-only state accessors.  The ``isValid()`` method is there to cover failures at pool shutdown when connections are being closed even as the pool is still being called by application threads.

Again, if an application holds onto an *unwrapped* Connection and tries to close it *anytime* after ``close()`` has been called to return the Connection to the pool, it is an application error.  If I were a user, I would certainly *want* a failure to occur.  I would probably fire the engineer who wrote it.